src/main/java/com/acquia/http/HMACMessageCreator.java

```diff
@@ -231,54 +263,66 @@ private String createMessage(String httpVerb, String host, String path, String q
         result.append(xAuthorizationTimestamp);
 
         //adding more if needed
-        byte[] requestBodyBytes = this.convertInputStreamIntoBtyeArray(requestBody);
-        if (this.isPassingRequestBody(httpVerb, requestBodyBytes)) {
+        if (this.isPassingRequestBody(contentLength, xAuthorizationContentSha256, requestBody)) {
             result.append("\n").append(contentType.toLowerCase());
-
-            //calculate body hash
-            byte[] encBody = DigestUtils.sha256(requestBodyBytes);
-            String bodyHash = Base64.encodeBase64String(encBody); //v2 specification requires base64 encoded SHA-256
-            result.append("\n").append(bodyHash);
+            result.append("\n").append(xAuthorizationContentSha256);
         }
         return result.toString();
     }
 
     /**
-     * Convert InputStream into byte[]
+     * Escape String with UTF-8 encoding
      * 
-     * @param inputStream
+     * @param theString
+     * @return
+     * @throws UnsupportedEncodingException
+     */
+    private String escapeProper(String theString) throws UnsupportedEncodingException {
+        return URLEncoder.encode(theString, ENCODING_UTF_8).replace("+", "%20");
+    }
+
+    /**
+     * Method to help check if requestBody needs to be passed or can be omitted
+     * 
+     * @param xAuthorizationContentSha256
+     * @param contentLength
      * @return
      * @throws IOException 
      */
-    private byte[] convertInputStreamIntoBtyeArray(InputStream inputStream) throws IOException {
-        if (inputStream == null) {
-            return null;
+    private boolean isPassingRequestBody(int contentLength, String xAuthorizationContentSha256,
```

Comment:
Consider splitting this method into two (isPassingRequestBody and getRequestBodySha356)